homeassistant/components/amcrest/switch.py

```diff
@@ -69,22 +69,22 @@ def available(self) -> bool:
         """Return True if entity is available."""
         return self._api.available
 
-    def turn_on(self, **kwargs: Any) -> None:
+    async def async_turn_on(self, **kwargs: Any) -> None:
         """Turn the switch on."""
-        self._turn_switch(True)
+        await self._turn_switch(True)
 
-    def turn_off(self, **kwargs: Any) -> None:
+    async def async_turn_off(self, **kwargs: Any) -> None:
         """Turn the switch off."""
-        self._turn_switch(False)
+        await self._turn_switch(False)
 
-    def _turn_switch(self, mode: bool) -> None:
+    async def _turn_switch(self, mode: bool) -> None:
```

Comment:
```suggestion
    async def _async_turn_switch(self, mode: bool) -> None:
```

Please prefix all async functions with `async_`